optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java

```diff
@@ -95,18 +77,29 @@ public ScoreDirectorFactoryFactory(ScoreDirectorFactoryConfig config) {
 
     protected AbstractScoreDirectorFactory<Solution_, Score_> decideMultipleScoreDirectorFactories(
             ClassLoader classLoader, SolutionDescriptor<Solution_> solutionDescriptor) {
-        EasyScoreDirectorFactory<Solution_, Score_> easyScoreDirectorFactory =
-                buildEasyScoreDirectorFactory(solutionDescriptor);
-        AbstractConstraintStreamScoreDirectorFactory<Solution_, Score_> constraintStreamScoreDirectorFactory =
-                buildConstraintStreamScoreDirectorFactory(solutionDescriptor);
-        IncrementalScoreDirectorFactory<Solution_, Score_> incrementalScoreDirectorFactory =
-                buildIncrementalScoreDirectorFactory(solutionDescriptor);
-        DroolsScoreDirectorFactory<Solution_, Score_> droolsScoreDirectorFactory = buildDroolsScoreDirectorFactory(
-                classLoader, solutionDescriptor);
+        // Load all known Score Director Factories via SPI.
+        ServiceLoader<ScoreDirectorFactoryProvider> scoreDirectorFactoryProviders =
+                ServiceLoader.load(ScoreDirectorFactoryProvider.class);
+        Map<ScoreDirectorType, AbstractScoreDirectorFactory<Solution_, Score_>> factories =
```

Comment:
Code style: Can we rename this variable to scoreDirectorFactoryMap?
It should end with Map, so it's clear that it's not an array (or list etc)